deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvStages.java

```diff
@@ -109,9 +109,16 @@ public void update(@Context SecurityContext sc,
         } catch (Exception e) {
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, e.toString());
         }
-        if (origBean.getStage_type() != EnvType.DEFAULT && origBean.getStage_type() != environBean.getStage_type()) {
-            throw new TeletaanInternalException(Response.Status.BAD_REQUEST, "Modification of stage type is not allowed!");
+
+        if (environBean.getStage_type() == null) {
```

Comment:
if (environBean.getStage_type() != null && origBean.getStage_type() != EnvType.DEFAULT && origBean.getStage_type() != environBean.getStage_type())

Reply:
If you leave environBean's stage_type to null, `normalizeEnvRequest` called in `updateStage` will fill the value with the default value, unintentionally change the value.

Reply:
ah i see. makes sense. so you are implicitly backfilling. can you do at the same place where the default is getting set?
I also think the problem is with the caller. since it's sending PATCH request using PUT API

Reply:
Yeah I agree it's the problem with the caller. But unfortunately we don't have a single PATCH and been using PUT APIs as PATCH. It will involve too many changes to do the PATCH conversion. 
I can't do at the same place where the default is getting, `public void updateStage(EnvironBean updateBean, String operator)` as it's a "PUT" style API. 

This is inside `normalizeEnvRequest()`
```java
        if (envBean.getStage_type() == null) {
            envBean.setStage_type(Constants.DEFAULT_STAGE_TYPE);
        }
```

Reply:
ok. I think we can still go ahead with the fix just to mitigate this issue incase someone misuses PUT for PATCH but should submit another fix in Rodimus(caller) to use PUT api in a correct way not necessarily replace with PATCH.

Reply:
Thanks. Just a clarification, all callers in deploy_board use this API in the PATCH fashion. This is not used in Rodimus.